agora/Agora/Utils.hs

```diff
@@ -262,46 +260,76 @@ paddValue = phoistAcyclic $
       )
 
 -- | Sum of all value at input.
-pvalueSpent :: Term s (PTxInfo :--> PValue)
+pvalueSpent :: Term s (PBuiltinList (PAsData PTxInInfo) :--> PValue)
 pvalueSpent = phoistAcyclic $
-  plam $ \txInfo' ->
-    pmatch txInfo' $ \(PTxInfo txInfo) ->
-      pfoldr
-        # plam
-          ( \txInInfo' v ->
-              pmatch
-                (pfromData txInInfo')
-                $ \(PTxInInfo txInInfo) ->
-                  paddValue
-                    # pmatch
-                      (pfield @"resolved" # txInInfo)
-                      (\(PTxOut o) -> pfromData $ pfield @"value" # o)
-                    # v
-          )
-        # pconstant mempty
-        # (pfield @"inputs" # txInfo)
+  plam $ \inputs ->
+    pfoldr
+      # plam
+        ( \txInInfo' v ->
+            pmatch
+              (pfromData txInInfo')
+              $ \(PTxInInfo txInInfo) ->
+                paddValue
+                  # pmatch
+                    (pfield @"resolved" # txInInfo)
+                    (\(PTxOut o) -> pfromData $ pfield @"value" # o)
+                  # v
+        )
+      # pconstant mempty
+      # inputs
 
 -- | Find the TxInInfo by a TxOutRef.
-pfindTxInByTxOutRef :: Term s (PTxOutRef :--> PTxInfo :--> PMaybe PTxInInfo)
+pfindTxInByTxOutRef :: Term s (PTxOutRef :--> PBuiltinList (PAsData PTxInInfo) :--> PMaybe PTxInInfo)
```

Comment:
Why all the switches from `PTxInfo` to `PBuiltinList (PAsData PTxInInfo)`?

Reply:
Passing the whole thing is inefficient.